src/parser/core/modules/DisengageGcds.tsx

```diff
@@ -0,0 +1,65 @@
+import {Trans} from '@lingui/react'
+import {Action} from 'data/ACTIONS'
+import {Event} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {Data} from 'parser/core/modules/Data'
+import {SimpleStatistic, Statistics} from 'parser/core/modules/Statistics'
+import React, {ReactNode} from 'react'
+
+export class DisengageGcds extends Analyser {
+	static override handle = 'disengagegcds'
+
+	@dependency data!: Data
+	@dependency statistics!: Statistics
+
+	private disengageGcds: number = 0
+
+	// Pass in the GCD you want to track as an Action.
+	protected trackedAction: Action = this.data.actions.UNKNOWN
```

Comment:
Probably better for a generic class like this to be marked as abstract and then to mark required items like this as abstract.  See BuffWindow or Procs for examples (I think Procs is probably the closer example for this here)

Also wondering if this should take an Action[] (array) instead, to allow for jobs that have multiple disengage GCDs they'd want to count, but I don't have a good mental image of what that would look like in the suggestion output.

Reply:
All right, I'll take a look at procs and update.

It's a good question about passing an array in rather than a single action. I can't think of any jobs where we'd want to aggregate multiple disengage attacks together and provide a single statistic as output. 

Some examples I can think of that have different ranged options would be:

1. Paladin, where they have both a ranged attack and hardcasting Holy Spirit as an option if they disengage. In that case we'd probably want to treat them differently, AFAIK preferring to hardcast Holy Spirit.
2. Ninja being able to use Raiton from range, but then maybe being short on mudras for their burst - I'm betting this would be more usefully analysed by pointing out if it causes them to miss uses of recommended actions in buff window
3. Reaper being able to use Harpe or Harvest Moon - Harvest Moon is better, hardcasting Soulsow in combat to get it is bad, is good, they already have a module that evaluates Harvest Moon usage

For the jobs I know about where there are multiple disengage options that we may want to track, it'd probably be better to implement two different instances of a class which have different info text and icons. Can you think of any classes where we may want to combine output?

Reply:
I've updated the class and overrides but left this supporting a single action rather than an array since the discussion here didn't come up with any examples that'd benefit from multiple ranged actions having the same info text: https://discord.com/channels/441414116914233364/441424599310270464/1269323101703901279 

It seems that generally when melee/tanks have multiple disengage actions one is preferrable to another so we'd want to give different feedback for each.